lib/ext/adviser.rb

```diff
@@ -0,0 +1,19 @@
+require 'csv'
+
+module AdviserExtension
+  extend ActiveSupport::Concern
+
+  class_methods do
+    def to_csv(_ = {})
+      CSV.generate do |csv|
+        csv << ['Ref. Number', 'Name', 'Firm']
+
+        find_each do |adviser|
+          csv << [adviser.reference_number, adviser.name, adviser.firm.registered_name]
+        end
+      end
+    end
+  end
+end
+
+Adviser.send(:include, AdviserExtension)
```

Comment:
@tomas-stefano the include is added outside of the `Adviser` class definitions because when it was added to the actual `Adviser` class in the **mas_rad-core** repo you advised that it should be moved out. 

Here is the changes that was proposed to the **mad_rad-core** repo (https://github.com/moneyadviceservice/mas-rad_core/pull/163/files). 

If we use a plain old ruby class to wrap the CSV functionality in what way could it open up the `Adviser` model for further addition of CSV-rendering implementations?

Is this proposal which you suggested the best way to go about this?

```ruby
class AdvisersListCsv
   def initialize(advisers)
      @advisers = advisers
   end

   def self.to_csv(_ = {})
     CSV.generate do |csv|
        csv << ['Ref. Number', 'Name', 'Firm']
 
        @advisers.each do |adviser|
          csv << [adviser.reference_number, adviser.name, adviser.firm.registered_name]
        end
     end
   end
end
```